### DISCUSSION TITLE
How to read the data in memray

### DISCUSSION THREAD

**Discussion by benhgm** [bloomberg/memray#232] [Q&A] [answered]
Hello there, I would like to use memray, but I am confused over how it works. I have some questions here:
1. What do the green, yellow and white cells in the flamegraph represent?
2. For checking memory leaks, I noticed that the values for the report with --leaks flag set and without it are the same. How do I make sense of this information?
Thank you so much for any advice!

**Comment by pradyunsg** [comment]
1. The colors are based on the extension of the file that contains the line of code pointed to, by that frame. Green is py, Yellow is c/cpp, Grey is "something else".

2. Did you run your application with the PYTHONMALLOC=malloc environment variable set? https://bloomberg.github.io/memray/flamegraph.html#memory-leaks-view

**Reply by benhgm** [reply]
Thanks @pradyunsg for the advice. I ran my code as follows:
```
PYTHONMALLOC=malloc memray run --trace-python-allocators script.py
``` 
Is this correct? I may have made a mistake

**Comment by pablogsal** [comment]
Also, did you had a chance to check out the docs regarding this topic: https://bloomberg.github.io/memray/flamegraph.html

**Comment by pablogsal** [comment, marked as answer]
> 2\. For checking memory leaks, I noticed that the values for the report with --leaks flag set and without it are the same. How do I make sense of this information?

That means that whatever was on the high watermark has been there until the end of the program. As we recommend in several places in the docs, please run with `--trace-python-allocators` or with the `PYTHONMALLOC=malloc` environment variable.

**Reply by benhgm** [reply]
Hi @pablogsal thanks for replying. Yes I did run with PYTHONMALLOC=malloc memray --trace-python-allocators

One noob question, must I run with export PYTHONMALLOC=malloc?

Thank you!

**Reply by pablogsal** [reply]
> One noob question, must I run with export PYTHONMALLOC=malloc?

Likely not because that will affect all processes in your machine (as long as they inherit the environment variable)  and that has the side-effect of making everything slower 😉
